### PR TITLE
Django+wagtail upgrades

### PR DESCRIPTION
**Pull request by rlskoeser**
updates ppa to python 3.11, django 5.0, and wagtail 5.2

adds developer docs about updating the wagtail test fixture

all unit tests passing locally and in CI

### CODE REVIEW

File: DEVELOPERNOTES.rst [hunk 1]
```diff
@@ -49,14 +51,36 @@ For more details, see `stack overflow <https://stackoverflow.com/questions/50168
...
+To update this to reflect changes in new versions of wagtail:
+
+1. Create an empty database to use for migrated the fixture.
+2. Check out a version of the codebase before any new migrations have been applied,
+and run migrations up to that point on the new database (`python manage.py migrate`)
+3. Remove preloaded wagtail content from the database using python console or web interface.
+4. Check out the new version of the code with the updated version of wagtail.
+5. Run migrations.
+6. Exported the migrated fixture data back to the fixture file. It's essential
+to use the `--natural-foreign` option::
```
Comment (quadrismegistus) [inline DEVELOPERNOTES.rst:81]: Nice, great to have this documented

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[tool.ruff]
+# allow longer line length
+line-length = 95
+
+# don't be as strict with formatting in migrations
+extend-exclude = ["ppa/*/migrations/*.py"]
```
Comment (quadrismegistus) [inline pyproject.toml:6]: interesting

File: ppa/pages/migrations/0008_contributor_page.py [hunk 3]
```diff
@@ -3,8 +3,8 @@
 from __future__ import unicode_literals
 
 import django.db.models.deletion
-import wagtail.core.blocks
-import wagtail.core.fields
+import wagtail.blocks
+import wagtail.fields
```
Comment (quadrismegistus) [inline ppa/pages/migrations/0008_contributor_page.py:7]: such a simple but pervasive change!

**Review by quadrismegistus — APPROVED** [review]
Looks great! Great documentation on the tricky process of navigating the wagtail ships to upgraded port.

File: pyproject.toml (hunk 2 above)
Comment (rlskoeser) [reply pyproject.toml:6]: it was complaining about long line length (but not fixing it automatically), and I couldn't figure out how to skip just that rule!

**Pull request merged**